src/Core/Services/StateService.cs

```diff
@@ -1708,6 +1710,21 @@ public async Task SetShouldConnectToWatchAsync(bool shouldConnect, string userId
                 ReconcileOptions(new StorageOptions { UserId = userId }, await GetDefaultStorageOptionsAsync());
             var key = Constants.ShouldConnectToWatchKey(reconciledOptions.UserId);
             await SetValueAsync(key, shouldConnect, reconciledOptions);
+            await UpdateLastUserShouldConnectToWatchAsync(shouldConnect);
+        }
+
+        public async Task<bool> GetLastUserShouldConnectToWatchAsync()
+        {
+            var reconciledOptions = await GetDefaultStorageOptionsAsync();
+            var key = Constants.LastUserShouldConnectToWatchKey;
+            return await GetValueAsync<bool?>(key, reconciledOptions) ?? false;
+        }
+
+        public async Task UpdateLastUserShouldConnectToWatchAsync(bool? shouldConnect = null)
```

Comment:
Should be `SetLastUserShouldConnectToWatchAsync` (instead of `Update...`)

Reply:
mmm I use `Update` because if the caller doesn't pass a value they are not setting anything but calling the method to update its internal value as it sees fit (in this case calling `GetLastUserShouldConnectToWatchAsync()`).
But if you still prefer `Set...` just so all the methods are the same is fine with me and I can change it.

Reply:
Going for `Set...`